jetty-jaspi/src/main/java/org/eclipse/jetty/security/jaspi/BasicAuthConfigFactory.java

```diff
@@ -0,0 +1,111 @@
+//
+// ========================================================================
+// Copyright (c) 1995-2021 Mort Bay Consulting Pty Ltd and others.
+//
+// This program and the accompanying materials are made available under the
+// terms of the Eclipse Public License v. 2.0 which is available at
+// https://www.eclipse.org/legal/epl-2.0, or the Apache License, Version 2.0
+// which is available at https://www.apache.org/licenses/LICENSE-2.0.
+//
+// SPDX-License-Identifier: EPL-2.0 OR Apache-2.0
+// ========================================================================
+//
+
+package org.eclipse.jetty.security.jaspi;
+
+import java.util.LinkedHashMap;
+import java.util.Map;
+import javax.security.auth.AuthPermission;
+
+import jakarta.security.auth.message.config.AuthConfigFactory;
+import jakarta.security.auth.message.config.AuthConfigProvider;
+import jakarta.security.auth.message.config.RegistrationListener;
+
+/** 
+ * A very basic {@link AuthConfigFactory} that allows for registering providers programmatically.
+ */
+public class BasicAuthConfigFactory extends AuthConfigFactory
```

Comment:
I'm not a fan of this naming as it can be confused to have some relation to Basic authentication. Can this be renamed to something like `DefaultAuthConfigFactory` (along with the .xml file, and the .mod file)?

Reply:
renamed